common/ext/class.ExtensionsManager.php

```diff
@@ -74,15 +66,7 @@ class common_ext_ExtensionsManager extends ConfigurableService
      */
     public static function singleton()
     {
-        if (! isset(self::$instance)) {
-            if (ServiceManager::getServiceManager()->has(self::SERVICE_ID)) {
-                self::$instance = ServiceManager::getServiceManager()->get(self::SERVICE_ID);
-            } else {
-                self::$instance = new common_ext_ExtensionsManager();
-                ServiceManager::getServiceManager()->propagate(self::$instance);
-            }
-        }
-        return self::$instance;
+        return ServiceManager::getServiceManager()->get(self::class);
```

Comment:
How can you ensure that the `ExtensionManager` is already exist in the `ServiceManager`?

Reply:
It is the first service we register:
https://github.com/oat-sa/generis/blob/master/common/oatbox/install/Installer.php#L51